### PR TITLE
Feat/implement chunks and render distance zappy gui

### PR DESCRIPTION
**Pull request by Thomaltarix**
I added the chunk mechanism.
A chunk is considered as a simple tile, its content, and the things that are on it.
If the chunk is not in the render distance radius it's not displayed.

Here's a little preview of my work :wink: :

https://github.com/FppEpitech/Zappy/assets/114673563/fb1619a0-ebb5-4f4c-9f3a-c9122500ca28

### CODE REVIEW

**Review by theophile-jr — COMMENTED** [review]
Fix the tests and it will LGTM :)

File: gui/src/Render/Render.cpp [hunk 1]
```diff
@@ -262,37 +323,31 @@ void Gui::Render::displayCursor()
...
-
-void Gui::Render::setCameraTile(std::pair<std::size_t, std::size_t> pos)
-{
-    _camera.setTilePos(pos);
-}
-
-std::pair<std::size_t, std::size_t> Gui::Render::getCameraTile() const
-{
-    return _camera.getTilePos();
-}
```
[the remark this answers is not in the document]
Comment (Thomaltarix) [reply gui/src/Render/Render.cpp:293]: I didn't deleted them, just changing the order of the method of the cpp file according the the order of the hpp one

File: gui/src/Render/Decoration.cpp
[the remark this answers is not in the document]
Comment (Thomaltarix) [reply gui/src/Render/Decoration.cpp:1]: I'm testing :+1:

[the remark this answers is not in the document]
Comment (Thomaltarix) [reply gui/src/Render/Decoration.cpp:1]: image

Well, i'll let them :+1:

**Review by mathieurobert1 — DISMISSED** [review]
Nice work 👍

File: gui/src/Render/Render.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (mathieurobert1) [reply gui/src/Render/Render.cpp:293]: I'm blind 🙈

**Review by mathieurobert1 — APPROVED** [review]
👍

**Review by theophile-jr — APPROVED** [review]

**Pull request merged**